packages/sdk/src/lend/namespaces/WalletLendNamespace.ts

```diff
@@ -1,83 +1,156 @@
 
-import type { SupportedChainId } from '@/constants/supportedChains.js'
-import { VerbsLendNamespace } from '@/lend/namespaces/VerbsLendNamespace.js'
 import type { LendProvider } from '@/lend/provider.js'
+import type { Asset } from '@/types/asset.js'
 import type {
   BaseLendConfig,
-  LendOptions,
-  LendTransaction,
+  ClosePositionParams,
+  GetPositionParams,
+  LendMarketId,
+  LendMarketPosition,
+  LendOpenPositionParams,
 } from '@/types/lend.js'
+import type { SmartWallet } from '@/wallet/base/SmartWallet.js'
+import type { Wallet } from '@/wallet/base/Wallet.js'
 
 /**
  * Wallet Lend Namespace
  * @description Full lending operations available on wallet.lend
  */
 export class WalletLendNamespace<
   TConfig extends BaseLendConfig = BaseLendConfig,
-> extends VerbsLendNamespace<TConfig> {
+> {
   constructor(
-    provider: LendProvider<TConfig>,
-    private readonly address: Address,
-  ) {
-    super(provider)
+    protected readonly provider: LendProvider<TConfig>,
+    private readonly wallet: Wallet,
+  ) {}
+
+  get config(): TConfig {
+    return this.provider.config
   }
 
+  // Inherited methods from VerbsLendNamespace
+  getMarkets = (...args: Parameters<LendProvider<TConfig>['getMarkets']>) =>
+    this.provider.getMarkets(...args)
+
+  getMarket = (...args: Parameters<LendProvider<TConfig>['getMarket']>) =>
+    this.provider.getMarket(...args)
+
+  supportedChainIds = (
+    ...args: Parameters<LendProvider<TConfig>['supportedChainIds']>
+  ) => this.provider.supportedChainIds(...args)
+
   /**
-   * Lend assets to a vault
-   * @description Will be renamed to execute() in the future
+   * Open a lending position
+   * @description Signs and sends a lend transaction from the wallet for the given amount and asset
    */
-  async lendExecute(
-    asset: Address,
-    amount: bigint,
-    chainId: SupportedChainId,
-    marketId?: string,
-    options?: LendOptions,
-  ): Promise<LendTransaction> {
-    // Set receiver to wallet address if not specified
-    const lendOptions: LendOptions = {
-      ...options,
-      receiver: options?.receiver || this.address,
+  async openPosition(params: LendOpenPositionParams): Promise<Hash> {
+    const lendOptions = {
+      ...params.options,
+      receiver: this.wallet.address,
+    }
+
+    const lendTransaction = await this.provider.openPosition({
+      amount: params.amount,
+      asset: params.asset,
+      marketId: params.marketId,
+      options: lendOptions,
+    })
+
+    const { transactionData } = lendTransaction
+    if (!transactionData) {
+      throw new Error('No transaction data returned from lend provider')
     }
 
-    return this.provider.lend(asset, amount, chainId, marketId, lendOptions)
+    // TODO Harry, can we pull sendBatch and send into the Wallet class so I can remove this?
+    if (!this.isSmartWallet(this.wallet)) {
+      throw new Error(
+        'Transaction execution is only supported for SmartWallet instances',
+      )
+    }
```

Comment:
@tremarkley  can we pull sendBatch and send into the Wallet class so I can remove this type guard?

Reply:
hmm, yeah we can figure out a way to share this. For a non-4337 wallet we cant batch txs but we could just call them sequentially